app/pb_kits/playbook/pb_currency/currency.rb

```diff
@@ -2,99 +2,57 @@
 
 module Playbook
   module PbCurrency
-    class Currency < Playbook::PbKit::Base
-      PROPS = %i[configured_align
-                 configured_classname
-                 configured_data
-                 configured_id
-                 configured_label
-                 configured_value
-                 configured_unit
-                 configured_dollar_sign
-                 configured_size].freeze
+    class Currency
+      include Playbook::Props
+      include ActionView::Helpers::NumberHelper
 
-      def initialize(align: default_configuration,
-                     classname: default_configuration,
-                     data: default_configuration,
-                     id: default_configuration,
-                     label: default_configuration,
-                     value: default_configuration,
-                     unit: default_configuration,
-                     dollar_sign: default_configuration,
-                     size: default_configuration)
-        self.configured_align = align
-        self.configured_classname = classname
-        self.configured_data = data
-        self.configured_id = id
-        self.configured_label = label
-        self.configured_value = value
-        self.configured_unit = unit
-        self.configured_dollar_sign = dollar_sign
-        self.configured_size = size
-      end
+      partial "pb_currency/currency"
 
-      def align
-        align_options = %w[left center right]
-        one_of_value(configured_align, align_options, "left")
-      end
+      prop :align, type: Playbook::Props::Enum,
+                   values: %w[left center right],
+                   default: "left"
+      prop :label
+      prop :amount, required: true
+      prop :size, type: Playbook::Props::Enum,
+                  values: %w[lg sm],
+                  default: "sm"
+      prop :unit, default: "$"
 
-      def dollar_sign
-        pb_dollar_sign = Playbook::PbBody::Body.new(classname: "dollar_sign", color: "light") do
-          default_value(configured_dollar_sign, "")
-        end
-        ApplicationController.renderer.render(partial: pb_dollar_sign, as: :object)
+      def classname
+        generate_classname("pb_currency_kit")
```

Comment:
looks like this needs to be passed `align` as well to match the last implementation.

Reply:
@garettarrowood  For some reason, my git ammend wasn't working correctly. You should be seeing the align_class private method I added now.